### PR TITLE
Catch service running exception

### PR DESCRIPTION
**Pull request by ChrisBr**
and propose reloading page.

### CODE REVIEW

File: src/api/app/controllers/webui/package_controller.rb [hunk 1]
```diff
@@ -390,8 +393,9 @@ def set_file_details
       end
     rescue ActiveXML::Transport::Error => e
       # TODO crudest hack ever!
-      if e.summary == 'service in progress'
+      if e.summary == 'service in progress' && @expand == 1
         @expand = 0
+        @service_running = true
```
Comment (mschnitzer) [inline src/api/app/controllers/webui/package_controller.rb]: Why is this important? I don't see this variable being used inside the view nor this controller (at least not after line 398).

Comment (ChrisBr) [inline src/api/app/controllers/webui/package_controller.rb]: ``set_file_details`` is a helper method which gets called in the ``show`` action, see line ``68``.

Comment (mschnitzer) [inline src/api/app/controllers/webui/package_controller.rb]: Ah ok, didn't see the `return`. Approved :-)

Comment (mschnitzer) [inline src/api/app/controllers/webui/package_controller.rb]: Ah ok, didn't see the `return`. Approved :-)

Comment (ChrisBr) [inline src/api/app/controllers/webui/package_controller.rb]: @mschnitzer JFYI the return doesn't matter here as I assign an instance variable which makes it also accessible in ``show``

File: src/api/app/controllers/webui/package_controller.rb [hunk 2]
```diff
@@ -65,8 +65,11 @@ def show
 
     @is_current_rev = false
     if set_file_details
-      if @forced_unexpand.blank?
+      if @forced_unexpand.blank? && @service_running.blank?
         @is_current_rev = !@revision || (@revision == @current_rev)
+      elsif @service_running
```
Comment (bgeuken) [inline src/api/app/controllers/webui/package_controller.rb]: Minor thing. You could move this condition up and then remove the ```@service_running.blank?``` in line 68.

Also I just realize that there is a small mismatch between ``` @service_running.blank? ``` and ``` @service_running ```. The first one seems to care about being empty, the latter not.

And line 398 indicates it's a boolean. Maybe we should just treat it that way;-)

**Comment by bgeuken** [comment]
Codewise it looks good to me. Can't say much about the functionality, but it sounds reasonalbe.

File: src/api/app/controllers/webui/package_controller.rb (hunk 2 above)
Comment (ChrisBr) [inline src/api/app/controllers/webui/package_controller.rb]: You're right! Will change that!

**Comment by ChrisBr** [comment]
Regarding functionality, it was requested in #2598. The main problem that you get an error flash is already fixed (was a backend 'problem'). 
@adrianschroeter ^^

**Comment by ChrisBr** [comment]
@adrianschroeter what is the opinion of PO?

**Comment by hennevogel** [comment]
LGTM

**Pull request merged**